### PR TITLE
merge branch 100_jupyterbook into main

### PR DESCRIPTION
**Pull request by asemposki**
This PR will merge `100_jupyterbook` into `main`, to include the Jupyter Book in the main branch of the repository. It will also allow for testing of the Jupyter Book and PR---before merging, I will push an error in a notebook and test if GitHub Actions catches this error. I will then push the fix of the error and comment that this merge is ready to occur. After this comment, @jared321 will review the PR.

- [x] Add pushes on `develop`
- [x] Make new Action for testing the Book

### CODE REVIEW

**Comment by asemposki** [comment]
I see Codecov wants coverage of the try/except statements I wrote to handle the multiprocessing runtime issues we had. I can attempt to write some pytests for that.

**Comment by asemposki** [comment]
The Jupyter Book failed to build when I pushed a bug in one of the notebooks that prevented the import of one of Taweret's modules. I will re-instate that module now and push the fix.

**Comment by asemposki** [comment]
Excellent, the bug fix worked and the Jupyter Book is again building.

@jared321, anytime you want to review this PR, go ahead.

**Review by jared321 — COMMENTED** [review]
Made two commits of changes related to finishing up Issue #91.  This could be done because the official notebooks are no longer in the sphinx docs folder hierarchy.  @asemposki please review those changes, build the HTML and PDF docs locally, and carefully check the content to see if it is in the final state that you want.

I have reviewed the other changes that I am capable of reviewing.  I've left some questions/comments.  Please let me know if you need any clarifications.

I didn't review the notebooks at all aside from confirming that all the notebooks from sphinx have made their way into the book TOC.  If the content of the notebooks has not changed on this branch and you haven't done so already, you might want to do a diff (e.g., [nbdime](https://nbdime.readthedocs.io/en/latest/)) of the notebooks in the Jupyterbook folder with the old notebooks from the sphinx folder.

File: .github/workflows/publish_book.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Build & publish Jupyter book
+env:
+  CLONE_PATH:    ${{ github.workspace }}
+  EXAMPLES_BOOK: ${{ github.workspace }}/book/_build/html
+on:
+  push:
+    branches:
+      - 100_jupyterbook
```
[the remark this answers is not in the document]
Comment (asemposki) [reply .github/workflows/publish_book.yml:8]: Yes, I added the to-do list to the PR. :)

File: .github/workflows/setup_tox_github_action.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (asemposki) [reply .github/workflows/setup_tox_github_action.sh:1]: Yeah, I think we can remove this, as long as we are happy with the other one.

File: .github/workflows/publish_book.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.12.3"]
```
[the remark this answers is not in the document]
Comment (asemposki) [reply .github/workflows/publish_book.yml:23]: I think 3.12 should be fine. I think I tried 3.12.3 when something was not working, but it was not fixed by this change.

File: book/notebooks/LMM/comparative_posterior.png
[the remark this answers is not in the document]
Comment (asemposki) [reply book/notebooks/LMM/comparative_posterior.png:1]: I think John or Dan might have some comparison plots they use in their notebooks to benchmark results, but I will check this today and make sure. If not, I can remove these images.

**Comment by asemposki** [comment]
OK, I've reviewed everything here, and everything looks good from a cursory glance, except the couple of changes mentioned in the comments above, which I'll check this afternoon.

File: .github/workflows/publish_book.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jared321) [reply .github/workflows/publish_book.yml]: Branches updated and new test version of the action spun off at 7e89b120.  The new action passed as expected.

File: .github/workflows/publish_book.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jared321) [reply .github/workflows/publish_book.yml]: Version changed in 7e89b120.

File: .github/workflows/setup_tox_github_action.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jared321) [reply .github/workflows/setup_tox_github_action.sh]: Removed at c1e76b8a.  All actions passing as expected.

File: book/notebooks/LMM/comparative_posterior.png
[the remark this answers is not in the document]
Comment (asemposki) [reply book/notebooks/LMM/comparative_posterior.png:1]: OK, I checked this notebook and all of the figures are generated explicitly, so there should be no version control on that.

**Comment by asemposki** [comment]
I checked the notebooks, and I see that Dan's is still showing all verbose sampling, which is not necessary and only wastes time printing and clutters the interface for a reader in the Jupyter Book, so I will clean that up as well before we merge this in. I should be able to do that quickly now.

**Comment by jared321** [comment]
@asemposki Have we now attended to all issues raised during the review?  If so, then I will try to break/fix the Jupyter book as you did to confirm that the new PR-based action is failing as intended and review all our changes one last time.

**Comment by asemposki** [comment]
I checked the RTD and everything looks good right now. I am good with you breaking and fixing the notebooks now. :)

**Comment by asemposki** [comment]
OK great! I will merge this and then make a branch from `main` to work on the content of the Book a bit more before the end of day tomorrow. I will also update the SDK to v0.4, since we'll need that done too.

**Pull request merged**